lib/consul/async/process_handler.rb

```diff
@@ -9,6 +9,14 @@ class ProcessDoesNotExist < StandardError
     class ProcessHandler
       attr_reader :command, :sig_reload, :sig_term, :pid, :exit_status, :last_signal_sent, :reload_scheduled
       attr_writer :reload_scheduled
+
+      def restart
+        warn "Restart process with pid #{pid}"
+        kill
+        start
```

Comment:
Same here - what about using Process module?

Reply:
because ProcessHandler.kill do it for me !